spec/job_template_spec.rb

```diff
@@ -16,14 +16,25 @@
     let(:json) { {'extra_vars' => "{\"instance_ids\":[\"i-999c\"],\"state\":\"absent\",\"subnet_id\":\"subnet-887\"}"} }
     let(:post_result_body) { {:job => 1} }
     let(:json_with_limit) { json.merge('limit' => 'machine_name') }
+    let(:response_double) { instance_double("Faraday::Response", :body => post_result_body.to_json) }
 
     it "runs an existing job template with a limit" do
-      launch_url      = "#{raw_instance["url"]}launch/"
-      response_double = instance_double("Faraday::Response", :body => post_result_body.to_json)
+      launch_url = "#{raw_instance["url"]}launch/"
+      raw_instance = build(:response_instance, :limitable_job_template, :klass => described_class)
```

Comment:
You can create two test variables in line 7
```
 let(:raw_instance_ask_limit)    { build(:response_instance, :job_template, :klass => described_class, :ask_limit_on_launch => true) }
 let(:raw_instance_no_limit)     { build(:response_instance, :job_template, :klass => described_class, :ask_limit_on_launch => false) }
```
Then in two test cases assert the following
```
expect { described_class.new(api, raw_instance_ask_limit.to_json).launch(json_with_limit) }.to_not raise_error
```
and
```
expect { described_class.new(api, raw_instance_no_limit.to_json).launch(json_with_limit) }.to raise_error ArgumentError
```

Here we want to convert the hash to JSON to mimic the response from API call. It also removes the confusion of strings vs symbols. No conversion is needed in the base model.

If you don't like to create `let` blocks, you can choose to add trait in the factory. The result will be the same.

Reply:
Traits won't work, symbols won't work, as I told you here: https://github.com/ansible/ansible_tower_client_ruby/pull/119#discussion_r211287343
And you also mentioned strings vs symbols here: https://gitter.im/ManageIQ/manageiq?at=5b7adf72796f7b601d6b4dea

However, I solved that in an other way.